common/txmgr/types/orm.go

```diff
@@ -0,0 +1,95 @@
+package types
+
+import (
+	"context"
+	"time"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+// NEWTX, TX, TXATTEMPT will be converted from generic types to structs at a future date to enforce design and type checks
+type ORM[ADDR any, CHAINID any, HASH any, NEWTX any, R any, TX any, TXATTEMPT any, TXID any, TXMETA any] interface {
+	CheckEthTxQueueCapacity(fromAddress ADDR, maxQueuedTransactions uint64, chainID CHAINID, opts ...any) (err error)
+	CountUnconfirmedTransactions(fromAddress ADDR, chainID CHAINID, opts ...any) (count uint32, err error)
+	CountUnstartedTransactions(fromAddress ADDR, chainID CHAINID, opts ...any) (count uint32, err error)
+	CreateEthTransaction(newTx NEWTX, chainID CHAINID, opts ...any) (tx TX, err error)
+	DeleteInProgressAttempt(attempt TXATTEMPT, opts ...any) error
+	EthTransactions(offset, limit int, opts ...any) ([]TX, int, error)
```

Comment:
In the generic Orm, remove all name references to Eth.
The names should be generic

Reply:
Yeah, I was thinking that all at once on the last PR so I can just focus on the core changes now without worrying about tedious name fixes.